src/erlfdb.erl

```diff
@@ -2201,3 +2321,16 @@ flush_transaction_ready_messages({erlfdb_transaction, TxRef} = Tx) ->
     after 0 ->
         ok
     end.
+
+open_dist_key(Type, UidData) ->
+    {?MODULE, open, Type, UidData}.
+
+open_with_key(Key, ClusterFile) ->
+    case persistent_term:get(Key, undefined) of
+        undefined ->
+            Db = create_database(ClusterFile),
```

Comment:
is it guaranteed that the NIF resource returned from `erlfdb_nif:create_database/1` will always remain valid/useful? (Whether it's an external client thread or not?)

If the resource becomes invalid temporarily due to a network disruption, I think that's fine, as long as the same resource can be used again once the network disruption resolves. If there's ever a scenario where that resource becomes invalid _permanently_, we're now in a scenario where we need to clear/overwrite a persistent term, right?

If we think we ever need to do that, I'm wondering if it's possible for us to store a 1 word term in the persistent term (like an atom) instead... which would avoid a whole VM GC if we ever needed to change it.

Reply:
The database object cannot be invalid. The official bindings treat the structure returned by create_database similarly.

https://github.com/apple/foundationdb/blob/10bf6bbfcaf8ddc5214cc806e94edc21eb18f4bc/bindings/python/fdb/impl.py#L2141